core/io/resource_format_binary.cpp

```diff
@@ -924,10 +925,21 @@ void ResourceLoaderBinary::get_dependencies(Ref<FileAccess> p_f, List<String> *p
 	for (int i = 0; i < external_resources.size(); i++) {
 		String dep;
 		String fallback_path;
+		String base_path = external_resources[i].path;
 
+		String foreign_resource_id = "";
```

Comment:
Strings are default initialized to empty.
```suggestion
		String foreign_resource_id;
```